ydb/core/tx/schemeshard/schemeshard_build_index__progress.cpp

```diff
@@ -50,6 +54,161 @@ constexpr const char* Name(TIndexBuildInfo::EState state) noexcept {
+    ui32 RetryCount = 0;
+    ui32 RowsBytes = 0;
+
+    NDataShard::TUploadStatus UploadStatus;
+
+public:
+    TUploadSampleK(TString targetTable,
+                   const TIndexBuildInfo::TLimits& limits,
+                   const TActorId& responseActorId,
+                   ui64 buildIndexId,
+                   TIndexBuildInfo::TSampleK::TRows&& init)
+        : TargetTable(std::move(targetTable))
+        , ResponseActorId(responseActorId)
+        , BuildIndexId(buildIndexId)
+        , Init(std::move(init))
+    {
+        LogPrefix = TStringBuilder() << "TUploadSampleK: build index id: " << BuildIndexId << " ";
+        Limits.MaxUploadRowsRetryCount = limits.MaxRetries;
+        Y_ASSERT(!Init.empty());
+    }
+
+    static constexpr auto ActorActivityType() {
+        return NKikimrServices::TActivity::SAMPLE_K_UPLOAD_ACTOR;
+    }
+
+    void UploadStatusToMessage(NKikimrIndexBuilder::TEvUploadSampleKResponse& msg) {
+        msg.SetUploadStatus(UploadStatus.StatusCode);
+        NYql::IssuesToMessage(UploadStatus.Issues, msg.MutableIssues());
+    }
+
+    void Describe(IOutputStream& out) const noexcept override {
+        out << Debug();
+    }
+
+    TString Debug() const {
+        return TStringBuilder() << LogPrefix << UploadStatus.ToString();
+    }
+
+    void Bootstrap(const NActors::TActorContext& ctx) {
+        Rows = std::make_shared<NTxProxy::TUploadRows>();
+        Rows->reserve(Init.size());
+        std::array<TCell, 2> PrimaryKeys;
+        PrimaryKeys[0] = TCell::Make(ui32{0});
+        for (ui32 i = 1; auto& [_, row] : Init) {
+            RowsBytes += row.size();
+            PrimaryKeys[1] = TCell::Make(ui32{i++});
+            // TODO(mbkkt) we can avoid serialization of PrimaryKeys every iter
+            Rows->emplace_back(TSerializedCellVec{PrimaryKeys}, std::move(row));
+        }
+        Init = {}; // release memory
+        RowsBytes += Rows->size() * TSerializedCellVec::SerializedSize(PrimaryKeys);
+
+        Types = std::make_shared<NTxProxy::TUploadTypes>(3);
+        Ydb::Type type;
+        type.set_type_id(Ydb::Type::UINT32);
+        (*Types)[0] = {NTableIndex::NTableVectorKmeansTreeIndex::LevelTable_ParentIdColumn, type};
+        (*Types)[1] = {NTableIndex::NTableVectorKmeansTreeIndex::LevelTable_IdColumn, type};
+        type.set_type_id(Ydb::Type::STRING);
+        (*Types)[2] = {NTableIndex::NTableVectorKmeansTreeIndex::LevelTable_EmbeddingColumn, type};
+
+        Become(&TThis::StateWork);
+
+        Upload(ctx, false);
+    }
+
+private:
+    STFUNC(StateWork) {
+        switch (ev->GetTypeRewrite()) {
+            HFunc(TEvTxUserProxy::TEvUploadRowsResponse, Handle);
+            CFunc(TEvents::TSystem::Wakeup, HandleWakeup);
+            default:
+                LOG_E("StateWork unexpected event type: " << ev->GetTypeRewrite() << " event: " << ev->ToString());
+        }
+    }
+
+    void HandleWakeup(const NActors::TActorContext& ctx) {
+        LOG_D("Retry upload " << Debug());
```

Comment:
More common context should be on the left side of the logging message.
Here LogPrefix is not actually used as a prefix.